src/instanceInfoResultType.ts

```diff
@@ -0,0 +1,25 @@
+/* eslint-disable camelcase */
+export type InstanceInfoResult = {
+  client: {
+    s: {
+      url: string;
+    }
+  },
+  build: {
+    enterprise_module: boolean;
+    version?: string;
+    raw: {
+      buildEnvironment: {
+        target_arch: string | null;
+        target_os: string | null;
+      }
+    }
+  },
+  dataLake: {
+    isDataLake: boolean;
+  },
+  genuineMongoDB: {
+    isGenuine: boolean;
+    dbType: string;
+  }
+};
```

Comment:
Another way we could do this is add a typescript definitions file to the data-service itself. While this does help our code it doesn't ensure this can't change in the data-service module. Not too much we can do there though.
I think using the data-service's instance function in our telemetry is a bit overkill - it fetches all of the databases and collection info for example, which we don't use/need. Maybe at some point we update it to be a bit more like `mongosh` where we're only pulling what we need: https://github.com/mongodb-js/mongosh/blob/master/packages/service-provider-server/src/cli-service-provider.ts#L234 . Not related to this pr but I figure it's good to bring up.

Reply:
Yeah, would be nice to have types in data service. But since we were talking about getting rid of data service completely for this project, I put a temporary solution here to fix the linting warning.